File: backend/app/schemas/executor.py
from typing import Any

from pydantic import BaseModel

from app.domain.constants.executor_constants import ExecutorName


class ExecutorCatalogItem(BaseModel):
    executor: ExecutorName
    label: str
    description: str
    connector_type: str
    mvp: bool
    config_schema: dict[str, Any]


class ListExecutorsResponse(BaseModel):
    items: list[ExecutorCatalogItem]

File: frontend/src/types/executor.ts
export type ExecutorName =
  | "mongo_find_one"
  | "mongo_find_many"
  | "mongo_insert"
  | "mongo_update"
  | "mongo_delete"
  | "mongo_aggregate"
  | "http_request";

export type ExecutorCatalogItem = {
  executor: ExecutorName;
  label: string;
  description: string;
  connector_type: string;
  mvp: boolean;
  config_schema: Record<string, { type: string; required?: boolean; values?: string[] }>;
};

export type ListExecutorsResponse = {
  items: ExecutorCatalogItem[];
};
